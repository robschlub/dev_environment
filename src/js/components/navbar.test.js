import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from './navbar';

describe('Navbar', () => {
  test('No props', () => {
    const component = renderer.create(<Navbar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test('Label is used to name button', () => {
  //   const component = renderer.create(<Button label='hello' />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('Children are used to label button', () => {
  //   const component = renderer.create(<Button>hello</Button>);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('Label is used when label and children defined', () => {
  //   const component = renderer.create(<Button label="button label">button child</Button>);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('Button class is defined, but not used', () => {
  //   const component = renderer.create(<Button className="" label='hello' />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('Button class used without btn', () => {
  //   const component = renderer.create(<Button className="-test-class" label='hello' />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // test('href is used', () => {
  //   const component = renderer.create(<Button label="hello" href="/#" />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
