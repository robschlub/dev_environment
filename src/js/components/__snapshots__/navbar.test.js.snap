// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Make "About" item active 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md nav-responsive navbar-color"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt=""
        className="d-inline-block align-top nav__home-icon"
        height="30"
        src="/static/icon-lg.png"
        width="30"
      />
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/introduction"
          >
            Introduction
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/Introduction"
          >
            Single
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/ShapesAndCornersTest"
          >
            Multi
          </a>
        </li>
        <li
          className="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            Lesson
          </a>
          <div
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <a
              className="dropdown-item"
              href="/Lessons/Math/Introduction"
            >
              1 - Why study shapes?
            </a>
            <a
              className="dropdown-item"
              href="/Lessons/Math/Angle"
            >
              2 - Angles
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/Lessons/Math/ShapesAndCornersTest"
            >
              Test
            </a>
          </div>
        </li>
      </ul>
      <form
        className="form-inline"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="search"
        />
      </form>
    </div>
  </nav>
</div>
`;

exports[`Navbar Make "Introduction" item active 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md nav-responsive navbar-color"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt=""
        className="d-inline-block align-top nav__home-icon"
        height="30"
        src="/static/icon-lg.png"
        width="30"
      />
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="/introduction"
          >
            Introduction
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/about"
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/Introduction"
          >
            Single
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/ShapesAndCornersTest"
          >
            Multi
          </a>
        </li>
        <li
          className="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            Lesson
          </a>
          <div
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <a
              className="dropdown-item"
              href="/Lessons/Math/Introduction"
            >
              1 - Why study shapes?
            </a>
            <a
              className="dropdown-item"
              href="/Lessons/Math/Angle"
            >
              2 - Angles
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/Lessons/Math/ShapesAndCornersTest"
            >
              Test
            </a>
          </div>
        </li>
      </ul>
      <form
        className="form-inline"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="search"
        />
      </form>
    </div>
  </nav>
</div>
`;

exports[`Navbar No props 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md nav-responsive navbar-color"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt=""
        className="d-inline-block align-top nav__home-icon"
        height="30"
        src="/static/icon-lg.png"
        width="30"
      />
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/introduction"
          >
            Introduction
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/about"
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/Introduction"
          >
            Single
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/Lessons/Math/ShapesAndCornersTest"
          >
            Multi
          </a>
        </li>
        <li
          className="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            Lesson
          </a>
          <div
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <a
              className="dropdown-item"
              href="/Lessons/Math/Introduction"
            >
              1 - Why study shapes?
            </a>
            <a
              className="dropdown-item"
              href="/Lessons/Math/Angle"
            >
              2 - Angles
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/Lessons/Math/ShapesAndCornersTest"
            >
              Test
            </a>
          </div>
        </li>
      </ul>
      <form
        className="form-inline"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="search"
        />
      </form>
    </div>
  </nav>
</div>
`;
